doc/guide.rst

```diff
@@ -201,6 +201,21 @@ What this does is tell pyOptSparse that the key ``obj_name`` in the function ret
 For optimizers that can do multi-objective optimization (e.g. NSGA2), multiple objectives can be added.
 Optimizers that can only handle one objective enforce that only a single objective is added to the optimization description.
 
+Analytic Derivatives
+++++++++++++++++++++
+
+pyOptSparse can automatically compute derivatives of the objective and constraint functions using finite differences or the complex-step method (See implementation details `here <https://mdolab-pyoptsparse.readthedocs-hosted.com/en/latest/api/gradient.html>`__).
+If analytic derivatives are available, users can define them within the ``sens()`` function.
+Note that when supplying analytic derivatives, users need to make sure the keys in the ``sens()`` function are consistent with the keys of design variables (e.g. ``var_name``), constraints (e.g. ``equality_constraint``) and objectives (e.g. ``obj_name``).
+Based on these keys (or names) users can separately define the analytic derivatives using a nested dictionary ``funcsSens``.
```

Comment:
We should explicitly provide an example of the `funcSens` dictionary (again, naming is flexible) as it would really facilitate the user understanding. You also did not mention the input arguments of the `sens(x, funcs)` function,  which again have a specific signature - please add it.

Reply:
No problem. I will add it.